### PR TITLE
Quantize slice op

### PR DESCRIPTION
**Pull request by zuzg**
### PR types
 New features

### PR changes
OPs

### Describe
Quantization of slice operator.<br>
In Ernie model (`test_quant2_int8_ernie_mkldnn`) quantized 1 sliced op, what caused drop in latency by 0.595%.

### CODE REVIEW

**Review by wozna — DISMISSED** [review]
LGTM

**Comment by zuzg** [comment]
@lidanqing-intel @sfraczek please review

**Review by lidanqing-vv — DISMISSED** [review]
LGTM

**Comment by lidanqing-vv** [comment]
Hi, @baoachun Do you think this PR could be approved? This PR added new attributes with asExtra. Slice op is op without weights and its quantization is easier.

**Review by XieYunshen — DISMISSED** [review]
LGTM for `set_tests_properties(test_analyzer_ernie_int8 PROPERTIES TIMEOUT 120)`

File: paddle/fluid/framework/ir/mkldnn/cpu_quantize_pass_tester.cc [hunk 1]
```diff
@@ -784,6 +788,113 @@ TEST(CpuQuantizePass, reshapeBetweenNonQuantizedOp) {
                   added_nodes_count, 2.0f * 127);
 }
 
+static const std::initializer_list<std::string> variable_names_slice = {
+    "a", "w1", "b", "c", "d", "e", "f"};
```
Comment (sfraczek) [inline paddle/fluid/framework/ir/mkldnn/cpu_quantize_pass_tester.cc]: it seems you are not using 3 variables: `w1,e,f`

File: paddle/fluid/operators/slice_op.cc [hunk 2]
```diff
@@ -244,7 +244,7 @@ class SliceOpMaker : public framework::OpProtoAndCheckerMaker {
         "mkldnn_data_type",
         "(string, default \"float32\"). Data type of mkldnn kernel")
         .SetDefault("float32")
-        .InEnum({"float32", "bfloat16"})
+        .InEnum({"float32", "bfloat16", "int8"})
```
Comment (sfraczek) [inline paddle/fluid/operators/slice_op.cc]: The ordering has been usually `float32, int8, bfloat16`, so what do you think about using that order?
image

File: paddle/fluid/framework/ir/mkldnn/cpu_quantize_pass.cc [hunk 3]
```diff
@@ -676,6 +676,57 @@ void CPUQuantizePass::QuantizeReshape(Graph* graph) const {
...
+    // skip if should not be quantized
+    if (!platform::HasOpINT8DataType(slice_op->Op())) {
+      LogQuantizationDisabled(slice_op);
+      return;
+    }
+    GET_IR_NODE_FROM_SUBGRAPH(prev_op, prev_op, slice_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(next_op, next_op, slice_pattern);
+
+    // skip if prev op and next op is not quantized
+    if (!(IsOpDequantized(prev_op)) && !(IsOpQuantized(next_op))) {
```
Comment (sfraczek) [inline paddle/fluid/framework/ir/mkldnn/cpu_quantize_pass.cc]: I think one parenthesis would be enough but it's okay to have them too if you prefer.

**Review by sfraczek — APPROVED** [review]
LGTM!

**Review by wozna — APPROVED** [review]
LGTM

**Comment by wozna** [comment]
@XieYunshen   Could you please review again?

**Review by lidanqing-vv — APPROVED** [review]
LGTM

**Comment by lidanqing-vv** [comment]
@Aganlengzi Hi, could you please help approve this PR, this PR **does not add** any new attribute, just added one possible value for existing attribute: "mkldnn_data_type". 

It was been approved before by Xieyunshen https://github.com/PaddlePaddle/Paddle/pull/37630#pullrequestreview-818505274. But cause there is one more reviewe so developer updated code, so it need approval again. Later will avoid similar situations.

**Comment by Aganlengzi** [comment]
LGTM

**Comment by Aganlengzi** [comment]
LGTM

**Review by Aganlengzi — APPROVED** [review]

**Comment by lidanqing-vv** [comment]
@Aganlengzi Hi, sorry but this PR need XieYunshen's approval, you can see this in log. Thanks.
```
0. You must have one QA (XieYunshen(Recommend) or chalsliu) approval for setting parameter RUN_TYPE as EXCLUSIVE, DIST, NIGHTLY, EXCLUSIVE:NIGHTLY or DISTNIGHTLY, or setting TIMEOUT properties.
```

**Comment by Aganlengzi** [comment]
@XieYunshen 
@lidanqing-intel please wait.

**Review by XieYunshen — APPROVED** [review]
LGTM for `set_tests_properties(test_analyzer_ernie_int8 PROPERTIES TIMEOUT 120)`

**Pull request merged**